core/loadpoint.go

```diff
@@ -658,22 +658,20 @@ func (lp *Loadpoint) syncCharger() error {
 		return err
 	}
 
-	// some chargers (i.E. Easee in some configurations) disable themself to be able to switch phases
-	if !enabled && lp.enabled && !lp.phaseSwitchCompleted() {
-		return lp.charger.Enable(true) // enable charger
-	}
+	validState := lp.chargerUpdateCompleted() && lp.phaseSwitchCompleted()
 
-	if lp.chargerUpdateCompleted() {
+	if validState {
 		defer func() {
 			lp.enabled = enabled
 			lp.publish(keys.Enabled, lp.enabled)
 		}()
 	}
 
+	// #1: check charger logic, fix charger state if necessary (this might be obsolete)
```

Comment:
Definitely not obsolete.

Reply:
Passiert das wirklich? Denn in den Charger Implementierungen, die ich kenne, wird das auf Ebene des Chargers geregelt…

Reply:
Ja passiert. Haben wir alles gehabt. Der Charger hat ggf. einen Zustand intern gespeichert. Dieser ist disabled, aber auf der Physik fließt Strom. Dann sagt der Charger im selben Moment "isCharging() : true" und "Enabled(): false". Deshalb reicht es nicht aus einfach nur den Zustand des Loadpoints auf Enabled zu setzen, sondern auch den Charger nochmal zu enablen. Das führt auf der Physik dann zu keiner Änerungen (lädt ja bereits), aber der Zustand der Charger Implementierung innerhalb von evcc ist dann konsistent.

Reply:
Dass wir es gehabt haben weiß ich.
Aber ob das die richtige Stelle ist bezweifle ich.
Denn dem Grunde nach sollte das auf der Ebene des Chargers geregelt sein.
Dafür gibt es extra die Funktion verifyEnabled in helper.go.

Reply:
Dazu gab es damals auch lange Diskussion. TL;DR - Die Charger dürfen Enabled() nicht dynamisch vom Ladezustand abhängig machen. Ich würde die Diskussion jetzt auch ungern nochmal wiederholen. Ich fände es gut wenn wir es einfach so lassen können wie es ist.

Reply:
Dass an der Stelle der PV Timer zurückgesetzt wird, ist aus meiner Sicht das Argument dafür das zu lassen. Ansonsten könnte man den Fall es genauso gut im Charger behandeln.

Reply:
Naja - hier sagt man dem Kind erstmal nur dass man es beim Lügen erwischt hat :-) Aber ich hatte Dir ja schon zugestimmt, hab mich wohl missverständlich ausgedrückt.

Reply:
Dann markier das hier als resolved. Aus meiner Sicht alles gut.

Reply:
> Dann markier das hier als resolved. Aus meiner Sicht alles gut.

Hab ich doch schon lange...